### PR TITLE
[CI] mypy install missing stubs

### PR DESCRIPTION
**Pull request by AnirudhDagar**
This PR is a follow up to #4011 (see for more details) and fixes failing python type check CI #4009 by installing the missing stubs.

I've not reverted the change in #4011, let me know if that is recommended. Thank you :))

### CODE REVIEW

**Comment by datumbox** [comment]
@AnirudhDagar feel free to revert the ignoring to see if your update works and solve this properly.

Note that the [CI is complaining ](https://app.circleci.com/pipelines/github/pytorch/vision/8820/workflows/59446e24-dbb5-49a5-b7e7-78f34180cd8c/jobs/640855) because the change is not present on config.yml.in

File: .circleci/config.yml [hunk 1]
```diff
@@ -207,6 +207,7 @@ jobs:
             pip install --user --progress-bar off mypy
             pip install --user --progress-bar off --pre torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
             pip install --user --progress-bar off --editable .
+            mypy --install-types
```
Comment (NicolasHug) [inline .circleci/config.yml:210]: you'll need to edit config.yml.in instead and call `python .circleci/regenerate.py` (it's a weird system, yup ;)   )

**Comment by AnirudhDagar** [comment]
Thanks @datumbox, will fix that.

File: .circleci/config.yml (hunk 1 above)
Comment (pmeier) [reply .circleci/config.yml:210]: Should we have a workflow that checks that both are in sync?

Comment (NicolasHug) [reply .circleci/config.yml:210]: I think that's the role of `circleci_consistency`

Comment (datumbox) [reply .circleci/config.yml:210]: Currently we have a job called `circleci_consistency` which checks for that.

**Comment by AnirudhDagar** [comment]
We need to pass y, currently, the command sits and waits for that input. Any suggestions for the same?
`mypy --install-types` doesn't support a `-y` option to automatically do that.

**Comment by AnirudhDagar** [comment]
@datumbox could you please help me here. Thanks in advance!

Since the output of mypy --install-types raises an error the CI automatically fails with that, without waiting for it to be fixed by installing the required stubs below 

This is the output on my local machine but CI exits as soon as the error is raised.

```bash
~/Desktop/Work/vision fix-mypy ❯ yes | mypy --install-types                                                                           

torchvision/datasets/utils.py:204: error: Library stubs not installed for "requests" (or incompatible with Python 3.8)  [import]
        import requests
    ^
torchvision/datasets/utils.py:204: note: Hint: "python3 -m pip install types-requests"
torchvision/datasets/utils.py:204: note: (or run "mypy --install-types" to install all missing stub packages)
torchvision/datasets/utils.py:204: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 110 source files)

Installing missing stub packages:
/Users/gollum/anaconda3/envs/vision/bin/python -m pip install types-requests

Install? [yN]
Collecting types-requests
  Using cached types_requests-0.1.8-py2.py3-none-any.whl (22 kB)
Installing collected packages: types-requests
Successfully installed types-requests-0.1.8
```

**Comment by AnirudhDagar** [comment]
The other option to fix this is more specific here. We can do `pip install --user --progress-bar off types-requests` instead of `yes | mypy --install-types`. I'll update the PR to see if that works.

**Comment by datumbox** [comment]
@AnirudhDagar I didn't intervene because you are doing all the checks and workarounds I would be doing as well. I'm not too familiar with all the changes that mypy did on the last version yesterday (part of the reason why I wanted a quick fix on #4011).

It's unclear to me why it's not blocking for install but hopefully your [last modification](https://github.com/pytorch/vision/pull/4014/commits/6ab6d4ec95b104356a9780a5c76a889b824aa46f) will work. Let's see.

**Review by datumbox — APPROVED** [review]
LGTM, thanks!

**Comment by AnirudhDagar** [comment]
@datumbox Yeah I completely understand that a fast fix was required at the moment. Thank you :))

**Comment by NicolasHug** [comment]
Thanks @AnirudhDagar !

**Pull request merged**